src/history/createStateHistory.ts

```diff
@@ -46,35 +90,49 @@ const createStateHistory: StateHistoryFactory = compose.mixin(createEvented, {
 
 		set(this: StateHistory, path: string) {
 			const privateState = privateStateMap.get(this);
-			privateState.current = path;
-			privateState.browserHistory.pushState({}, '', path);
-			this.emit({
-				type: 'change',
-				value: path
-			});
+			const { base, browserHistory } = privateState;
+
+			const value = ensureLeadingSlash(path);
+			const fullPath = prependBase(base, path);
+
+			privateState.current = value;
+			browserHistory.pushState({}, '', fullPath);
+			this.emit({ type: 'change', value });
 		},
 
 		replace(this: StateHistory, path: string) {
 			const privateState = privateStateMap.get(this);
-			privateState.current = path;
-			privateState.browserHistory.replaceState({}, '', path);
-			this.emit({
-				type: 'change',
-				value: path
-			});
+			const { base, browserHistory } = privateState;
+
+			const value = ensureLeadingSlash(path);
+			const fullPath = prependBase(base, path);
+
+			privateState.current = value;
+			browserHistory.replaceState({}, '', fullPath);
+			this.emit({ type: 'change', value });
 		}
 	},
-	initialize(instance: StateHistory, { window }: StateHistoryOptions = { window: global }) {
+	initialize(instance: StateHistory, { base = '/', window }: StateHistoryOptions = { window: global }) {
+		if (base !== '/') {
+			if (/#/.test(base)) {
+				throw new TypeError('base must not contain \'#\'');
+			}
+			if (/\?/.test(base)) {
+				throw new TypeError('base must not contain \'?\'');
+			}
+		}
+
 		const { history: browserHistory, location } = window;
 
 		const privateState: PrivateState = {
-			current: location.pathname + location.search,
+			base,
+			current: stripBase(base, location.pathname + location.search),
 			browserHistory
 		};
 		privateStateMap.set(instance, privateState);
 
 		instance.own(on(window, 'popstate', () => {
-			const path = location.pathname + location.search;
+			const path = stripBase(base, location.pathname + location.search);
```

Comment:
I'm not sure what you mean by "use it in both locations"? This code runs after a `popstate` event so the browser's location will have changed. The `stripBase` call has more logic than just slicing the path string and that extra logic may be necessary here.